src/api-service/__app__/agent_can_schedule/__init__.py

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env python
+#
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+import azure.functions as func
+from onefuzztypes.enums import ErrorCode, TaskState
+from onefuzztypes.models import Error, NodeCommand, StopNodeCommand
+from onefuzztypes.requests import CanScheduleRequest
+from onefuzztypes.responses import CanSchedule
+
+from ..onefuzzlib.agent_authorization import verify_token
+from ..onefuzzlib.pools import Node, NodeMessage
+from ..onefuzzlib.request import not_ok, ok, parse_uri
+from ..onefuzzlib.tasks.main import Task
+
+
+def get(req: func.HttpRequest) -> func.HttpResponse:
+    request = parse_uri(CanScheduleRequest, req)
+    if isinstance(request, Error):
+        return not_ok(request, context="CanScheduleRequest")
+
+    current_version = ""
+
+    node = Node.get_by_machine_id(request.machine_id)
+    if not node:
+        return not_ok(
+            Error(code=ErrorCode.UNABLE_TO_FIND, errors=["unable to find node"]),
+            context=request.machine_id,
+        )
+
+    allowed = True
+    work_stopped = False
+    if node.version != current_version:
```

Comment:
Can this be a function of Node?  I imagine we'll likely want to do this in multiple places.

Reply:
moves logic to Node class